[PDR-640] Update PDR generators for partial TheBasics responses

pegbertsch opened the pull request:
## Resolves *[PDR-640](https://precisionmedicineinitiative.atlassian.net/browse/PDR-640)*


## Description of changes/additions
* Update PDR data generators to account for partial TheBasics payloads (e.g., for PROFILE_UPDATE)
* Fix pre-existing issue with resource generator UBR flag datatypes found during testing of new code

## Tests
- [x] unit tests
Existing unit tests updated, added new enrollment status calculator tests

pegbertsch committed:
[PDR-640] Update PDR generators to account for partial TheBasics responses

pegbertsch commented on rdr_service/dao/bq_questionnaire_dao.py:
```diff
@@ -61,7 +62,8 @@ def make_bqrecord(self, p_id, module_id, latest=False, convert_to_enum=False):
             inner join participant p on p.participant_id = qr.participant_id
             left join hpo h on p.hpo_id = h.hpo_id
-            where qr.participant_id = :p_id and qr.questionnaire_id IN (
+            -- Screen out classification DUPLICATE (1) responses from the PDR data [PDR-640]
```
@robabram Can you please verify that we should be filtering out all DUPLICATE classification `questionnaire_response` records from the `pdr_mod` table data?  This was not added after the addition of the original `is_duplicate` column

robabram replied:
Yes, we should be filtering out known duplicates here.  Good catch.

pegbertsch committed:
Add new enrollment status calculator unit test

pegbertsch committed:
More unit test changes

pegbertsch committed:
Remove classification_type field from BigQuery pdr_mod_* table data results
Limits PDR schema change; v_pdr_participant_module data will have classification_type and
can be joined to v_pdr_mod_* table to filter results
